blocks/algorithms/__init__.py

```diff
@@ -276,7 +278,7 @@ def compute_steps(self, gradients):
 
 
 class SteepestDescent(StepRule):
-    """A step in the direction opposite to the gradient.
+    """A step in the direction proportional to the gradient.
```

Comment:
Right, sorry, that's what you get for trying to read code while also following class.

Anyway, the argument still holds: `gradient` is confusing, because it's no longer the gradient if it's part of a chain. So I would rename the argument to `step` everywhere, the first step being the gradient as passed by the `GradientDescent` class.